pythonforandroid/bootstrap.py

```diff
@@ -257,6 +266,83 @@ def _unpack_aar(self, aar, arch):
+        # Todo: All the recipes using gnustl_shared should be migrated to use
+        # c++_shared because in ndk r18 it will be the only stl available:
+        # https://developer.android.com/ndk/guides/cpp-support
+        for recipe_name in self.ctx.recipe_build_order:
+            recipe = Recipe.get_recipe(recipe_name, self.ctx)
+            if recipe.stl_depends is None:
+                continue
+            if recipe.stl_depends not in ('c++_shared', 'gnustl_shared'):
+                raise BuildInterruptingException(
+                    'You have a non supported value in attribute stl_depends '
+                    '"{stl_lib}" for recipe "{recipe_name}". The'
+                    'value should be one of: "c++_shared" or "gnustl_shared"'.format(
+                        stl_lib=recipe.stl_depends, recipe_name=recipe.name))
+            if recipe.stl_depends not in libs_to_load:
+                libs_to_load.append(recipe.stl_depends)
+            if all(lib in libs_to_load for lib in ('c++_shared', 'gnustl_shared')):
+                break
+
+        # Add crystax if necessary
+        if self.ctx.python_recipe.from_crystax:
+            libs_to_load.append('crystax')
+
+        # Add libraries that python depends on
+        if 'sqlite3' in self.ctx.recipe_build_order:
+            libs_to_load.append('sqlite3')
+        if 'libffi' in self.ctx.recipe_build_order:
+            libs_to_load.append('ffi')
+        if 'openssl' in self.ctx.recipe_build_order:
+            recipe = Recipe.get_recipe('openssl', self.ctx)
+            libs_to_load.append('ssl' + recipe.version)
+            libs_to_load.append('crypto' + recipe.version)
+
+        # Add the bootstrap libs
+        [libs_to_load.append(i) for i in self.libraries_to_load]
+
+        # Add the corresponding python lib
+        python_version = self.ctx.python_recipe.major_minor_version_string
+        if python_version[0] == '3':
+            python_version += 'm'
+        libs_to_load.append('python' + python_version)
+
+        # Add libs that should be loaded after python because depends on it
+        if 'boost' in self.ctx.recipe_build_order:
+            # Todo: check boost loading dependencies and create a proper
+            # loading order. For now it works for libtorrent's recipe, the only
+            # recipe that depends on boost...so...for now it's fine, but should
+            # be enhanced, maybe using similar method than
+            # `pythonforandroid.build.copylibs_function`?
+            for l in listdir(self.ctx.get_libs_dir(arch.arch)):
+                if l.startswith('libboost_'):
+                    libs_to_load.append(l[3:-3])
+        if 'libtorrent' in self.ctx.recipe_build_order:
+            libs_to_load.append('torrent_rasterbar')
+        if 'libiconv' in self.ctx.recipe_build_order:
+            libs_to_load.append('iconv')
+        if 'libzbar' in self.ctx.recipe_build_order:
+            libs_to_load.append('zbar')
```

Comment:
It feels kinda weird to have these specific use cases in a core module rather than in a dedicated recipe. What do you think?